From 0b964a0292bef5ee056ed094d8384e7df7bba0db Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:38 -0400
Subject: [ethernet] mlx4: Select PTP_1588_CLOCK

Message-id: <8f8a897b70faccc4bfe200fff92e700bf29b4f43.1410425016.git.avadai@redhat.com>
Patchwork-id: 93608
O-Subject: [Patch RHEL7.1 V1 042/197] mlx4_en: Select PTP_1588_CLOCK
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Shawn Bohrer <sbohrer@rgmadvisors.com>

Now that mlx4_en includes a PHC driver it must select PTP_1588_CLOCK.

   drivers/built-in.o: In function `mlx4_en_get_ts_info':
>> en_ethtool.c:(.text+0x391a11): undefined reference to `ptp_clock_index'
   drivers/built-in.o: In function `mlx4_en_remove_timestamp':
>> (.text+0x397913): undefined reference to `ptp_clock_unregister'
   drivers/built-in.o: In function `mlx4_en_init_timestamp':
>> (.text+0x397b20): undefined reference to `ptp_clock_register'

Fixes: ad7d4eaed995d ("mlx4_en: Add PTP hardware clock")
Signed-off-by: Shawn Bohrer <sbohrer@rgmadvisors.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 74b9c3ea847f060c784e86453f1ad77dd05a7a8f)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/Kconfig b/drivers/net/ethernet/mellanox/mlx4/Kconfig
index eb520ab..563495d 100644
--- a/drivers/net/ethernet/mellanox/mlx4/Kconfig
+++ b/drivers/net/ethernet/mellanox/mlx4/Kconfig
@@ -6,6 +6,7 @@ config MLX4_EN
  tristate "Mellanox Technologies 10Gbit Ethernet support"
  depends on PCI
  select MLX4_CORE
+ select PTP_1588_CLOCK
  ---help---
    This driver supports Mellanox Technologies ConnectX Ethernet
    devices.
-- 
1.7.1